libs/blocks/global-navigation/blocks/search/gnav-search.js

```diff
@@ -18,24 +20,31 @@ const CONFIG = {
 
 class Search {
   constructor(config) {
-    this.label = config.label;
     this.icon = config.icon;
     this.trigger = config.trigger;
     // TODO: could this elem be passed through config?
-    this.parent = this.trigger.closest('.mainnav-wrapper');
+    this.parent = this.trigger.closest('.feds-nav-wrapper');
 
     this.init();
   }
 
-  init() {
+  async init() {
+    await this.getLabels();
     this.decorate();
     this.addEventListeners();
   }
 
+  async getLabels() {
+    const config = getConfig();
+    const sheet = 'feds';
+    this.labels = {};
+    [this.labels.search, this.labels.clearResults, this.labels.tryAdvancedSearch] = await replaceKeyArray(['search', 'clear-results', 'try-advanced-search'], config, sheet);
+  }
+
```

Comment:
Since we just use the function and variables once, IMO just having it all in the init is a bit cleaner - but it's only a slight preference, feel free to keep your code or use this suggestion:
```suggestion
  async init() {
    this. Labels = {};
    [this.labels.search, this.labels.clearResults, this.labels.tryAdvancedSearch] = await replaceKeyArray(['search', 'clear-results', 'try-advanced-search'], getConfig(), 'feds');
    this.decorate();
    this.addEventListeners();
  }
```

Reply:
I usually prefer the init method to just declare instance variables and call methods, so it's super-clear what's going on, without any sort of clutter. I think I'll leave it as-is for now and we can come back to it if we decide on another rule.